src/neo/Wallets/Wallet.cs

```diff
@@ -561,6 +563,7 @@ public long CalculateNetworkFee(DataCache snapshot, Transaction tx)
                     // Check verify cost
                     using ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, tx, snapshot.CreateSnapshot(), settings: ProtocolSettings);
                     engine.LoadContract(contract, md, CallFlags.ReadOnly);
+                    if (invocationScript != null) engine.LoadScript(invocationScript, configureState: p => p.CallFlags = CallFlags.None);
```

Comment:
Also, we don't need to forget about adding the invocation script's VarSize to `size` (5 lines above), because it may be non-empty now.

Reply:
which size?

Reply:
```suggestion
                    // Empty verification and non-empty invocation scripts
                    var invSize = (invocationScript != null ? invocationScript.GetVarSize() : Array.Empty<byte>().GetVarSize());
                    size += Array.Empty<byte>().GetVarSize() + invSize;
                    // Check verify cost
                    using ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, tx, snapshot.CreateSnapshot(), settings: ProtocolSettings);
                    engine.LoadContract(contract, md, CallFlags.ReadOnly);
                    if (invocationScript != null) engine.LoadScript(invocationScript, configureState: p => p.CallFlags = CallFlags.None);
```